Core/src/IfcGeometryConverter/FaceConverter.h

```diff
@@ -22,460 +22,648 @@
+						typename IfcEntityTypesT::IfcBSplineSurfaceForm surfaceForm = surface->SurfaceForm;
 
-						if(cylindrical_surface) {
-							// Get radius.
-							/*std::shared_ptr<typename IfcEntityTypesT::IfcPositiveLengthMeasure>& cylindrical_radius =
-								cylindrical_surface->Radius;*/
-							double cylindrical_radius = cylindrical_surface->Radius;
-							// TODO: implement		// Interpret values and calculate.
-						}
+						// Get attributes 5-7. For information only.
+						typename IfcEntityTypesT::IfcLogical uClosed = surface->UClosed;
+						typename IfcEntityTypesT::IfcLogical vClosed = surface->VClosed;
+						typename IfcEntityTypesT::IfcLogical selfIntersect = surface->SelfIntersect;
+					}
+					*/
 
-						// (2/4) IfcPlane SUBTYPE of IfcElementarySurface
-						std::shared_ptr<typename IfcEntityTypesT::IfcPlane> elementary_surface_plane =
-							std::dynamic_pointer_cast<typename IfcEntityTypesT::IfcPlane>(elementary_surface);
-
-						if(elementary_surface_plane) {
-							//  1----0     create big rectangular plane
-							//  |    |     ^ y
-							//  |    |     |
-							//  2----3     ---> x
-							{
-								double plane_span = HALF_SPACE_BOX_SIZE;
-								polyline_data->beginPolyline();
-								polyline_data->addVertex(elementary_surface_matrix * carve::geom::VECTOR(plane_span, plane_span, 0.0));
-								polyline_data->addVertex(elementary_surface_matrix * carve::geom::VECTOR(-plane_span, plane_span, 0.0));
-								polyline_data->addVertex(elementary_surface_matrix * carve::geom::VECTOR(-plane_span, -plane_span, 0.0));
-								polyline_data->addVertex(elementary_surface_matrix * carve::geom::VECTOR(plane_span, -plane_span, 0.0));
-
-								polyline_data->addPolylineIndex(0);
-								polyline_data->addPolylineIndex(1);
-								polyline_data->addPolylineIndex(2);
-								polyline_data->addPolylineIndex(3);
-							}
-							return;
-						}
+				}
 
-						// (3/4) IfcSphericalPlane SUBTYPE of IfcElementarySurface
-						std::shared_ptr<typename IfcEntityTypesT::IfcSphericalSurface> spherical_surface =
-							std::dynamic_pointer_cast<typename IfcEntityTypesT::IfcSphericalSurface>(elementary_surface);
+				/*! \brief  Converts \c IfcCurveBoundedPlane to ...
+					\param	surface	\c IfcCurveBoundedPlane entity to be interpreted.
+					\param	pos
+					\param	polylineData
+					\note	The \c IfcCurveBoundedPlane is a subtype of \c IfcBoundedSurface.
+				*/
 
-						if(spherical_surface) {
-							// Get radius.
-							/*std::shared_ptr<typename IfcEntityTypesT::IfcPositiveLengthMeasure>& spherical_radius =
-								spherical_surface->m_Radius;*/
+				void convertIfcCurveBoundedPlane(const EXPRESSReference<typename IfcEntityTypesT::IfcCurveBoundedPlane>& surface,
+					const carve::math::Matrix& pos,
+					std::shared_ptr<carve::input::PolylineSetData>& polylineData) {
 
-							double spherical_radius = spherical_surface->Radius;
+					const throw oip::UnhandledException(surface);

```

Comment:
```suggestion
					throw oip::UnhandledException(surface);
```
Similarly everywere else.

Reply:
Resolved: https://github.com/tumcms/Open-Infra-Platform/pull/151/commits/6b394b45649131a68ecf2829fc1d5eef2f606470